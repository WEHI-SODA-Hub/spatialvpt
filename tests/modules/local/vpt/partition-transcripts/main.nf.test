nextflow_process {

    name "Test Process PARTITION_TRANSCRIPTS"
    script "modules/local/vpt/partition-transcripts/main.nf"
    process "PARTITION_TRANSCRIPTS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/tests/data/cellpose_micron_space.parquet', checkIfExists: true) ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/detected_transcripts.csv', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
