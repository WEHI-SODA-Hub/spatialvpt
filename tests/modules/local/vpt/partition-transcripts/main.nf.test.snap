{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        [
                            "cell_by_gene_repartitioned.csv:md5,aceba0f6ed50361af265b04db6b8b7f8",
                            "detected_transcripts_repartitioned.csv:md5,6d4441b9fd1b8c4bfb6945d00f89c569"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,c6227a7da97aad53d158177b0b2f2722"
                ],
                "transcripts": [
                    [
                        {
                            "id": "small_set"
                        },
                        [
                            "cell_by_gene_repartitioned.csv:md5,aceba0f6ed50361af265b04db6b8b7f8",
                            "detected_transcripts_repartitioned.csv:md5,6d4441b9fd1b8c4bfb6945d00f89c569"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c6227a7da97aad53d158177b0b2f2722"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-08-05T16:15:06.899851245"
    }
}